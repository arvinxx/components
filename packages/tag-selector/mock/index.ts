import type { Item } from '../src/type';

export const smallLabel: Item[] = [
  {
    title: '未分组',
    key: '1526132530988',
    value: [
      {
        text: '测试标签1',
        key: 'rJecwftRf',
        refText: '21312',
      },
      {
        text: '测试标签2',
        key: 'rJecwfsdstRf',
        refText: '1352dx',
      },
    ],
  },
];
export const mediumLabel: Item[] = [
  {
    title: '未分组',
    key: '1526132530988',
    value: [
      {
        text: '测试标签1',
        key: 'rJecwftRf',
        refText: '21312',
      },
      {
        text: '测试标签2',
        key: 'rJecwfsdstRf',
        refText: '1352dx',
      },
    ],
  },
  {
    title: '性别',
    key: 'S1a3uMt0M',
    value: [
      {
        text: '男',
        key: 'Byb9PfYRG',
      },
      {
        text: '女',
        key: 'Byb9PfY2RG',
      },
    ],
  },
];
export const largeLabel: Item[] = [
  {
    title: '未分组',
    key: '1526132530988',
    value: [
      {
        text: '少门分，用被孟',
        key: 'rJecwftRf',
      },
      {
        text: '格活最任，问',
        key: 'SJQcwMYAz',
      },
      {
        key: 'r1H89JcRz',
        text: '百管百',
      },
      {
        key: 'ry2koFcAz',
        text: '离格活最',
      },
    ],
  },
  {
    title: '性别',
    key: 'S1a3uMt0M',
    value: [
      {
        text: '男',
        key: 'Byb9PfYRG',
      },
      {
        text: '女',
        key: 'Byb9PfY2RG',
      },
    ],
  },
  {
    title: '未的方式',
    key: 'Sy4JKMF0M',
    value: [
      {
        text: '约车，正各发存转论西月',
        key: 'BJhdPGt0M',
      },
      {
        text: '我柜把',
        key: 'BkjODGtAG',
      },
      {
        text: '门识二此里除青，由门之六料',
        key: 'HkJtwMFRM',
      },
    ],
  },
  {
    title: '未命发地方多少',
    key: 'ryIyKGKAM',
    value: [
      {
        text: '身门识二此里除',
        key: 'ry0twGtCM',
      },
      {
        text: '问出豆',
        key: 'rkrYvGtRz',
      },
      {
        text: '体而改几些地共类边，',
        key: 'HJetvGFRM',
      },
    ],
  },
  {
    title: '发多少',
    key: 'HJ%JtMKCz',
    value: [
      {
        text: 'D候我柜',
        key: 'r1FYPftAM',
      },
      {
        text: '节受三自验始候，',
        key: 'r1U9wzFRG',
      },
      {
        text: '各发存转论',
        key: 'HJjFvfF0M',
      },
    ],
  },
  {
    title: '未命发多少',
    key: 'H15kKMK0G',
    value: [
      {
        text: '节受三自验始候',
        key: 'BkwKDMKCz',
      },
      {
        text: '传门',
        key: 'Bk3twMKAf',
      },
    ],
  },
];
