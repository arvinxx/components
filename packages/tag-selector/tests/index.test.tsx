import React from 'react';
import { render } from '@testing-library/react';

import TagSelector from '@arvinxu/tag-selector';
import { smallLabel } from '../mock';

describe('TagSelector', () => {
  test('正常渲染', () => {
    const { container } = render(<TagSelector />);
    expect(container).toMatchSnapshot();
  });

  it('smallLabel', () => {
    const { container } = render(<TagSelector value={smallLabel} />);
    expect(container).toMatchSnapshot();
  });
});

// describe('响应操作', () => {
//   let Input;
//   beforeEach(() => {
//     Input = wrapper.find('Input');
//   });
//
//   it('newLabelOnInput should run when change', () => {
//     Input.simulate('change', { target: { value: '1' } });
//     expect(wrapper.state('newLabel')).toEqual('1');
//   });
//   it('newLabelOnFocus should run when focus', () => {
//     expect(wrapper.state('newLabelPlaceHolder')).toEqual('添加条目');
//     Input.simulate('focus');
//     expect(wrapper.state('newLabelPlaceHolder')).toEqual('');
//   });
//
//   describe('newLabelOnBlur', () => {
//     it("newLabelOnBlur should run when blur if there's content", () => {
//       wrapper.setState({ newLabel: '1' });
//       Input.simulate('blur');
//       expect(wrapper.state('newLabelPlaceHolder')).toEqual('添加条目');
//       expect(wrapper.state('newLabel')).toEqual('');
//       expect(dispatch.callCount).toEqual(1);
//     });
//     it("newLabelOnBlur should not run when blur if there's no content", () => {
//       Input.simulate('blur');
//       expect(wrapper.state('newLabelPlaceHolder')).toEqual('添加条目');
//       expect(wrapper.state('newLabel')).toEqual('');
//       expect(dispatch.callCount).toEqual(0);
//     });
//   });
//   describe('newLabelOnBlur', () => {
//     it("newLabelOnBlur should run when blur if there's content", () => {
//       wrapper.setState({ newLabel: '1' });
//       Input.simulate('pressEnter');
//       expect(wrapper.state('newLabelPlaceHolder')).toEqual('');
//       expect(wrapper.state('newLabel')).toEqual('');
//       expect(dispatch.callCount).toEqual(1);
//     });
//     it("newLabelOnPressEnter should not run when pressEnter if there's no content", () => {
//       Input.simulate('pressEnter');
//       expect(wrapper.state('newLabelPlaceHolder')).toEqual('');
//       expect(wrapper.state('newLabel')).toEqual('');
//       expect(dispatch.callCount).toEqual(0);
//     });
//   });
// });
