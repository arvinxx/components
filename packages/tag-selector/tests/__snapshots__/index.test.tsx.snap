// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`正常渲染 1`] = `
<div>
  <div
    class="avx-tag-selector-container"
  >
    <div
      class="avx-tag-selector-key-container"
    >
      <input
        class="ant-input avx-tag-selector-add-key"
        placeholder="添加条目"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
